shell/common/api/atom_api_native_image.cc

```diff
@@ -541,8 +542,21 @@ bool Converter<electron::api::NativeImage*>::FromV8(
   // Try converting from file path.
   base::FilePath path;
   if (ConvertFromV8(isolate, val, &path)) {
+    // Throw when path is invalid, i.e file doesn't exist
+    base::ThreadRestrictions::ScopedAllowIO allow_io;
```

Comment:
We should avoid operations on the main thread that involve adding `ScopedAllowIO` where possible.

Also it seems redundant to check for path to throw the bellow error as `!(*out)->image().IsEmpty()` already gives you whether an image could be initialized or not. Please use that check to throw the error.

Reply:
The issue there, though, is that it doesn't tell you _why_ the image couldn't be initialized, does it? My goal here is to make it clear to users when the fault lies with their path and not with a more oblique and unclear failure.

Reply:
Well whatever error you require can be obtained from `electron::api::NativeImage::CreateFromPath` since it has all the knowledge about the image and its path when the initialization happened, you can try to retrieve the info from there.

Reply:
Updated!